Source/Core/VideoBackends/OGL/OGLTexture.cpp

```diff
@@ -164,15 +149,73 @@ void OGLTexture::Bind(unsigned int stage)
   }
 }
 
-bool OGLTexture::Save(const std::string& filename, unsigned int level)
+std::optional<AbstractTexture::RawTextureInfo> OGLTexture::MapFullImpl()
+{
+  if (GLInterface->GetMode() != GLInterfaceMode::MODE_OPENGL)
+    return {};
+
+  m_staging_data.reserve(m_config.width * m_config.height * 4);
```

Comment:
Likely out of scope for now, but this staging region should be accessable by the GPU itself. So some map-able OpenGL buffer.